error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/StreamRules.java

```diff
@@ -247,12 +248,13 @@ Optional<S> after(Stream<T> stream, Function<? super T, S> function) {
   /** In order to test whether a stream has any element, simply try to find one. */
   static final class StreamIsEmpty<T> {
     @BeforeTemplate
-    boolean before(Stream<T> stream) {
+    boolean before(Stream<T> stream, Collector<? super T, ?, ? extends Collection<?>> collector) {
       return Refaster.anyOf(
           stream.count() == 0,
           stream.count() <= 0,
           stream.count() < 1,
-          stream.findFirst().isEmpty());
+          stream.findFirst().isEmpty(),
+          stream.collect(collector).isEmpty());
```

Comment:
couldn't you theoretically use a filtering collector though?

Reply:
Yes! We should add a comment calling out this caveat. I'll add a commit later this morning :+1:

Reply:
Added two more comments. Since this kind of is rare (and since we rewrite top-level `filtering` collectors to a regular `filter` step), I think this is an acceptable limitation for now.